litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go

```diff
@@ -88,3 +88,50 @@ func NewEvent(clusterEvent model.ClusterEventInput, r store.StateData) (string,
 
 	return "", errors.New("ERROR WITH CLUSTER EVENT")
 }
+
+// UpsterWorkFlowRun Updates or Inserts a new Workflow Run into the DB
+func UpsterWorkFlowRun(input model.WorkflowRunInput, r store.StateData) (string, error) {
+	cluster, err := cluster.VerifyCluster(*input.ClusterID)
+	if err != nil {
+		log.Print("ERROR", err)
+		return "", err
+	}
+	newWorkflowRun := model.WorkflowRun{
+		ClusterID:     cluster.ClusterID,
+		ClusterName:   cluster.ClusterName,
+		ProjectID:     cluster.ProjectID,
+		LastUpdated:   strconv.FormatInt(time.Now().Unix(), 10),
+		WorkflowRunID: input.WorkflowRunID,
+		WorkflowName:  input.WorkflowName,
+		ExecutionData: input.ExecutionData,
+		WorkflowID:    "000000000000",
+	}
+	subscriptions.SendWorkflowEvent(newWorkflowRun, r)
```

Comment:
```suggestion
	subscriptions.SendWorkflowEvent(model.newWorkflowRun(newWorkflowRun), r)
```
This can be converted like this after you move mongo struct from the gql struct